File: frontend/src/types/proposal.ts
/**
 * Proposal types aligned with backend schemas
 */

// Validation status
export type ValidationStatus = 'pending' | 'in_progress' | 'completed' | 'failed'

// Risk level
export type RiskLevel = 'low' | 'medium' | 'high' | 'critical'

// Proposal interface
export interface Proposal {
    id: string
    title: string
    description?: string
    contract_id?: string
    validation_status: ValidationStatus
    risk_level?: RiskLevel
    risk_score?: number
    validation_report: ValidationReport
    detected_clauses: DetectedClause[]
    compliance_checks: Record<string, any>
    created_by: string
    created_at: string
    updated_at?: string
    validated_at?: string
}

// Proposal list item
export interface ProposalListItem {
    id: string
    title: string
    validation_status: ValidationStatus
    risk_level?: RiskLevel
    risk_score?: number
    created_by: string
    created_at: string
    validated_at?: string
}

// Proposal creation
export interface ProposalCreate {
    title: string
    description?: string
    contract_id?: string
    upload_id?: string
}

// Proposal statistics
export interface ProposalStats {
    total: number
    pending: number
    in_progress: number
    completed: number
    failed: number
}

// Validation request
export interface ValidateContractRequest {
    contract_type?: string
    create_proposal?: boolean
    custom_checks?: string[]
}

// Validation issue
export interface ValidationIssue {
    severity: 'low' | 'medium' | 'high' | 'critical'
    message: string
    location?: string
    suggestion?: string
}

// Detected clause
export interface DetectedClause {
    name: string
    description: string
    location?: string
    risk_level?: string
}

// Validation report
export interface ValidationReport {
    issues?: ValidationIssue[]
    suggestions?: string[]
    clauses?: DetectedClause[]
    compliance?: Record<string, any>
    raw_analysis?: string
    error?: string
}

// Full validation response
export interface ValidationResponse {
    proposal_id?: string
    risk_score: number
    risk_level: string
    issues: ValidationIssue[]
    suggestions: string[]
    clauses: DetectedClause[]
    compliance: Record<string, any>
    raw_analysis?: string
}

// Status display helpers
export const VALIDATION_STATUS_LABELS: Record<ValidationStatus, string> = {
    pending: 'Pending',
    in_progress: 'In Progress',
    completed: 'Completed',
    failed: 'Failed'
}

export const VALIDATION_STATUS_COLORS: Record<ValidationStatus, string> = {
    pending: 'bg-gray-100 text-gray-800 dark:bg-gray-800 dark:text-gray-200',
    in_progress: 'bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-200',
    completed: 'bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-200',
    failed: 'bg-red-100 text-red-800 dark:bg-red-900 dark:text-red-200'
}

export const RISK_LEVEL_LABELS: Record<RiskLevel, string> = {
    low: 'Low Risk',
    medium: 'Medium Risk',
    high: 'High Risk',
    critical: 'Critical Risk'
}

export const RISK_LEVEL_COLORS: Record<RiskLevel, string> = {
    low: 'bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-200',
    medium: 'bg-yellow-100 text-yellow-800 dark:bg-yellow-900 dark:text-yellow-200',
    high: 'bg-orange-100 text-orange-800 dark:bg-orange-900 dark:text-orange-200',
    critical: 'bg-red-100 text-red-800 dark:bg-red-900 dark:text-red-200'
}

File: backend/app/schemas/proposal.py
"""
Proposal Pydantic schemas for request/response validation.
"""
from pydantic import BaseModel, Field, UUID4
from typing import Optional, List, Dict, Any
from datetime import datetime
from app.db.models.proposal import ValidationStatus, RiskLevel


class ProposalBase(BaseModel):
    """Base proposal schema."""
    title: str = Field(..., min_length=1, max_length=500)
    description: Optional[str] = None


class ProposalCreate(ProposalBase):
    """Schema for creating a new proposal."""
    contract_id: Optional[UUID4] = None
    upload_id: Optional[UUID4] = None


class Proposal(ProposalBase):
    """Schema for proposal response."""
    model_config = {"from_attributes": True}
    
    id: UUID4
    contract_id: Optional[UUID4] = None
    validation_status: ValidationStatus
    risk_level: Optional[RiskLevel] = None
    risk_score: Optional[float] = None
    validation_report: Dict[str, Any] = {}
    detected_clauses: List[Dict[str, Any]] = []
    compliance_checks: Dict[str, Any] = {}
    created_by: UUID4
    created_at: datetime
    updated_at: Optional[datetime] = None
    validated_at: Optional[datetime] = None


class ProposalListItem(BaseModel):
    """Schema for proposal list item."""
    model_config = {"from_attributes": True}
    
    id: UUID4
    title: str
    validation_status: ValidationStatus
    risk_level: Optional[RiskLevel] = None
    risk_score: Optional[float] = None
    created_by: UUID4
    created_at: datetime
    validated_at: Optional[datetime] = None


class ProposalStats(BaseModel):
    """Schema for proposal statistics."""
    total: int
    pending: int
    in_progress: int
    completed: int
    failed: int


class ValidateContractRequest(BaseModel):
    """Schema for contract validation request."""
    contract_type: Optional[str] = None
    create_proposal: bool = True
    custom_checks: List[str] = Field(default_factory=list)


class ValidationIssue(BaseModel):
    """Schema for validation issue."""
    severity: str
    message: str
    location: Optional[str] = None
    suggestion: Optional[str] = None


class DetectedClause(BaseModel):
    """Schema for detected clause."""
    name: str
    description: str
    location: Optional[str] = None
    risk_level: Optional[str] = None


class ValidationReport(BaseModel):
    """Schema for validation report response."""
    proposal_id: Optional[UUID4] = None
    risk_score: float
    risk_level: str
    issues: List[ValidationIssue] = []
    suggestions: List[str] = []
    clauses: List[DetectedClause] = []
    compliance: Dict[str, Any] = {}
    raw_analysis: Optional[str] = None
